src/core/useBVH.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react'
+import { Mesh, Object3D } from 'three'
+import { acceleratedRaycast, computeBoundsTree, disposeBoundsTree } from 'three-mesh-bvh'
+
+export interface BVHOptions {
+  splitStrategy?: 'CENTER' | 'AVERAGE' | 'SAH'
+  verbose?: boolean
+  setBoundingBox?: boolean
+  maxDepth?: number
+  maxLeafTris?: number
+}
+
+export function useBVH(object3D: React.MutableRefObject<Mesh | undefined>, options?: BVHOptions) {
+  const ref = React.useRef<Object3D>()
+
+  React.useEffect(() => {
+    if (object3D.current) {
+      object3D.current.raycast = acceleratedRaycast
+      let geometry: any = object3D.current.geometry
+      geometry.computeBoundsTree = computeBoundsTree
+      geometry.disposeBoundsTree = disposeBoundsTree
+      geometry.computeBoundsTree(options)
+
+      return () => {
+        if (geometry.boundsTree) {
+          geometry.disposeBoundsTree()
+        }
+      }
+    }
+  }, [object3D, options])
```

Comment:
afaik the ref we're passing won't cause a re-render of any kind? i.e if you change what the ref is attached too the effect won't re-run. If we want to do that, we should return a `useCallback` styled hook like [this](https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node).

Reply:
@joshuaellis So for the ref I totally agree, but if we use useCallback here I think it will make the feature more complex to use? Maybe I did not understand, I'm talking about ref={useBVH} here.
Also, I think we need useEffect to enable BVH onMount. If you think that useCallback is easier I'm totally ok to change to that, I just want to call "computeBoundsTree" once the component is mounted. About the unMount we don't really care as anyway the geometry will get disposed.

Reply:
Because using `useRef` won't cause a re-render, so the `useEffect` hook won't refire when the dependancy changes cause it's a ref, instead we should look at using the `useCallback` pattern:

```js
export function useBVH(options) {
  const ref =  React.useCallback(
    (node) => {
      node.raycast = acceleratedRaycast
      let geometry: any = node.geometry
      geometry.computeBoundsTree = computeBoundsTree
      geometry.disposeBoundsTree = disposeBoundsTree
      geometry.computeBoundsTree(options)
    },
    [options]
  )

  return ref
}
```

So you'd pass the ref returned from the function to the `jsx` component.

Reply:
@joshuaellis well I tested a bit and I don't think the useCallback approach would work in this usecase. The useEffect approach is working and re-render when needed, I can enable/disable BVH if required, and even dynamic options works as you can see in this example: https://drei-git-fork-renaudrohlinger-bvh-hook-pmndrs.vercel.app/?path=/story/performance-usebvh--default-story.

I update the PR so you can merge it.
But if you really think that useCallback is better I would be happy to learn from it but I'm pretty sure it will get called multiple times unnecessarily and we won't be able to use any hook such as useHelper.

Reply:
No no, it was just an idea, if it's re-rendering correctly lets keep your method! ⭐  Thanks for investigating it though 😄

Reply:
Ah I see, thanks for the heads up 😊